Allow passing a function for lazy-showvalues (#182)
timholy/ProgressMeter.jl: README.md

end
```

> In the above example `showvalues` keyword is evaluated in each iteration, regardless
> if the progress will be reprinted or not. For avoiding that and prevent overhead a
> zero-argument function can be passed that returns the `showvalues` object.

Reviewer: ```suggestion
In the above example, the data passed to `showvalues` is evaluated even if the progress bar is not updated. To avoid this unnecessary computation and reduce the overhead you can alternatively pass a zero-argument function as a callback to the `showvalues` keyword.
```

```julia

> **x,n = 1,10

Reviewer: A typo I guess?

```suggestion
x,n = 1,10
```

p = Progress(n)
generate_showvalues(iter, x) = () -> [(:iter,iter), (:x,x)]
for iter = 1:10
    x *= 2
    sleep(0.5)

>     ProgressMeter.next!(p; showvalues = generate_showvalues(iter, x))

Reviewer: ```suggestion
# unlike `showvalues=generate_showvalues(iter, x)()`, this version only evaluate the function when necessary
ProgressMeter.next!(p; showvalues = generate_showvalues(iter, x))
```
